pkgs/development/libraries/jpeg-xl/default.nix

```diff
@@ -0,0 +1,69 @@
+{ stdenv, lib, fetchgit
+, cmake
+, giflib
+, gperftools
+, gtest
+, libjpeg
+, libpng
+, libwebp
+, openexr
+, pkgconfig
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+
+  pname = "jpeg-xl";
+  version = "2020-10-29-8f2e0857";
+
+  src = fetchgit { # as of writing, `fetchFromGitLab` cannot fetch submodules
+    # Temporary workaround for https://gitlab.com/wg1/jpeg-xl/-/issues/57
+    url = "https://gitlab.com/nh21/jpeg-xl";
+    # url = "https://gitlab.com/wg1/jpeg-xl";
+    rev = "8f2e08572e686b8016e1efac0a5eb5bc7fca1b81";
+    sha256 = "0i58vr1jmawma4jck3v2ym76adksknj8jasnxvfj96zs2q5y025i";
+    # There are various submodules in `third_party/`.
+    fetchSubmodules = true;
+    # The `CMakeLists.txt` errors on absence of `.git` directories.
```

Comment:
Can't we patch the cmakelists?

Reply:
Would that be better? There may be legitimate use cases for keeping `.git`, e.g. if the build system uses it to determine `--version` output.

Reply:
If I remember this correctly ``.git`` is most of the time stripped because it can be really large and is not  really deterministic and the version is patched or passed in via a flag.

Reply:
> If I remember this correctly `.git` is most of the time stripped because it can be really large and is not really deterministic and the version is patched or passed in via a flag. I think patching the version is better than keeping .git.

OK, then we need to patch it. I currently don't have time to look into it though (probably OK given that it's a draft PR anyway).

Reply:
This seems to be addressed in the current version of `jpeg-xl`, it no longer errors on absence of `.git` directories.